### ISSUE TITLE
multi tracking

### ISSUE THREAD

**Issue by RustoMCSpit** [Leonidius20/RecordingStudio#19]
is it possible to have this work like a 4 track recorder where i can layer voice recordings on different lanes. you have all the technology to do this, youd just need to copy and paste ui elements and have a pop up telling the user to wear headphones to prevent audio bleeding into other tracks or muting the tracks while recording

**Comment by RustoMCSpit** [comment]
youd also have to save the tracks separately and together which means collecting them all in a single folder.

**Comment by Leonidius20** [comment]
Hello! Thank you for the suggestion.
I may implement something like this either in this app, or in another one built on top of this one's technology, but later on

**Comment by RustoMCSpit** [comment]
![1732189089404316531686037082300](https://github.com/user-attachments/assets/1393dc56-de8a-4de3-a9bc-7332e35ae22a)

**Comment by RustoMCSpit** [comment]
if this is done, id recommend allowing users to upload audio files for tracks from files or copy and paste

**Comment by RustoMCSpit** [comment]
> Hello! Thank you for the suggestion. I may implement something like this either in this app, or in another one built on top of this one's technology, but later on

id recommend just having it in this app and having a plus button appear under the audio wave. having two apps doubles your workload

**Comment by RustoMCSpit** [comment]
also, making this compatible with dawproject would allow for immediare import into daws https://github.com/bitwig/dawproject

**Comment by brownwatchman** [comment]
I totally second this! I tried working on a 4track recorder of my own but never could quite figure it out. Maybe you could take some ideas from it. I believe that a simple 4 track option with the feature to bounce to the master track and have your 4 tracks cleared to continue to build and bounce again would be ideal.  Essentially, it would be taking inspiraron from both the J4T app, the Zoom R4 and a classic tape 4 track recorder.  

Here is a mock up I did; 

![Screenshot_20250114_225438_Figma](https://github.com/user-attachments/assets/806ef147-7741-45cc-8ed6-efcc313682a7)

**Comment by RustoMCSpit** [comment]
problem with that design is that youre limited to four track exclusively. what if i wanted 5 track?

i think apple's voice memos app does this extremely sleekly but the above design is very pretty.

**Comment by brownwatchman** [comment]
The "5th" track would be the master track. Again, you could record 4 tracks, bounce them to the master track which would clear up the 4 tracks again. Record another 4 track, bounce them to the master track etc. This would be unlimited,  based on your device storage. You could essentially build a full track with 50 tracks if you wanted to. Check out the Zoom R4, which has this in place to give you a better idea of what I'm referring to. 

Essentially,  the master track would host all of the bounces you made, including effects, panning, and levels.

**Comment by RustoMCSpit** [comment]
okay but thatd be entirely destructive. bouncing tracks and merging them as one because there is no "add track" button is a very bad way to go about this

**Comment by brownwatchman** [comment]
You're right, it is meant to be a commitment, which I previously stated it was. Again, my purpose for the idea if for it to be a modern 4 track recorder. If you understand the history of the 4 track recorder, you would know that recording and bouncing are essential to building a track that requires more than 4 tracks. 

If you want it to be a multitrack recorder, then yeah, you're essentially talking about a DAW, which are similar, but different in workflow. In your initial comment, you stated that you wanted this to be 4 track recorder, which the idea that I provided is. Hope this clarifies.

**Comment by RustoMCSpit** [comment]
i just used a four track as an example to explain. regardless, your design is great!
